<launch>

    <!-- keypoint parameters -->
    <arg name="up_left_u" default="234"/>
    <arg name="up_left_v" default="150"/>
    <arg name="down_left_u" default="192"/>
    <arg name="down_left_v" default="370"/>
    <arg name="down_right_u" default="560"/>
    <arg name="down_right_v" default="370"/>
    <arg name="up_right_u" default="492"/>
    <arg name="up_right_v" default="150"/>
    <arg name="h_min" default="0"/>
    <arg name="h_max" default="180"/>
    <arg name="s_min" default="0"/>
    <arg name="s_max" default="100"/>
    <arg name="v_min" default="150"/>
    <arg name="v_max" default="255"/>
    <arg name="hough_threshold" default="100"/>
    <arg name="min_line_length" default="80"/>
    <arg name="max_line_gap" default="10"/>
    <arg name="angle_diff_threshold" default="0.01"/>
    <arg name="min_distance_limit" default="10"/>
    <arg name="max_distance_limit" default="20"/>

    <!-- transform parameters -->
    <arg name="up_left" default="250"/>
    <arg name="transform_width" default="150"/>
    <arg name="transform_height" default="150"/>

    <arg name="show_image" default="false"/>
    <arg name="line_position_v_threshold" default="240"/>

    <node pkg="stop_line_detector" type="stop_line_detector" name="stop_line_detector" ns="recognition">
        <param name="UP_LEFT_U" value="$(arg up_left_u)"/>
        <param name="UP_LEFT_V" value="$(arg up_left_v)"/>
        <param name="DOWN_LEFT_U" value="$(arg down_left_u)"/>
        <param name="DOWN_LEFT_V" value="$(arg down_left_v)"/>
        <param name="DOWN_RIGHT_U" value="$(arg down_right_u)"/>
        <param name="DOWN_RIGHT_V" value="$(arg down_right_v)"/>
        <param name="UP_RIGHT_U" value="$(arg up_right_u)"/>
        <param name="UP_RIGHT_V" value="$(arg up_right_v)"/>
        <param name="UP_LEFT" value="$(arg up_left)"/>
        <param name="TRANSFORM_WIDTH" value="$(arg transform_width)"/>
        <param name="TRANSFORM_HEIGHT" value="$(arg transform_height)"/>
        <param name="H_MIN" value="$(arg h_min)"/>
        <param name="H_MAX" value="$(arg h_max)"/>
        <param name="S_MIN" value="$(arg s_min)"/>
        <param name="S_MAX" value="$(arg s_max)"/>
        <param name="V_MIN" value="$(arg v_min)"/>
        <param name="V_MAX" value="$(arg v_max)"/>
        <param name="HOUGH_THRESHOLD" value="$(arg hough_threshold)"/>
        <param name="MIN_LINE_LENGTH" value="$(arg min_line_length)"/>
        <param name="MAX_LINE_GAP" value="$(arg max_line_gap)"/>
        <param name="ANGLE_DIFF_THRESHOLD" value="$(arg angle_diff_threshold)"/>
        <param name="MIN_DISTANCE_LIMIT" value="$(arg min_distance_limit)"/>
        <param name="MAX_DISTANCE_LIMIT" value="$(arg max_distance_limit)"/>
        <param name="SHOW_IMAGE" value="$(arg show_image)"/>
        <param name="LINE_POSITION_V_THRESHOLD" value="$(arg line_position_v_threshold)"/>
    </node>

</launch>
